pages/store/index.vue

```diff
@@ -130,6 +146,18 @@ const shouldLoadMore = useElementVisibility(infiniteScrollDetectorElement)
 const { handleError } = useErrorHandler()
 const isMobile = useMediaQuery('(max-width: 768px)')
 
+const authorName = computed(() => getRouteQuery('author', ''))
+const publisherName = computed(() => getRouteQuery('publisher', ''))
+
+const isSearchMode = computed(() => !!searchQuery.value)
+
+// Search query key for bookstore store
+const searchQuery = computed(() => {
+  if (authorName.value) return `author:${authorName.value}`
+  if (publisherName.value) return `publisher:${publisherName.value}`
+  return ''
+})
```

Comment:
Maybe order computed chronologically
```suggestion
// Search query key for bookstore store
const searchQuery = computed(() => {
  if (authorName.value) return `author:${authorName.value}`
  if (publisherName.value) return `publisher:${publisherName.value}`
  return ''
})

const isSearchMode = computed(() => !!searchQuery.value)
```